- Importing libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from scipy.stats import chi2_contingency

In [3]:
#Importing .csv file
data = pd.read_csv(r'C:\Users\GOKUL BABU K\Downloads\Sales_add.csv')

In [4]:
# Checking the data imported properly
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [6]:
#Checking for the null values
data.isnull().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

- Null values are not present the data

In [7]:
#Finding the overview of the data
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

- Sample size is 22 which is less that 30 and we need to do a one tailed two sample T-test.


# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

- Ho = Mean sales before and after the digital marketing are same
- Ha = Mean sales after the digital marketing is more than before
- Confidance interval 0.05

In [8]:
#Checking the column names
data.columns

Index(['Month', 'Region', 'Manager', 'Sales_before_digital_add(in $)',
       'Sales_After_digital_add(in $)'],
      dtype='object')

In [23]:
# Finding T value and P value of the data.
tvalue, pvalue=stats.ttest_rel(data['Sales_After_digital_add(in $)'],data['Sales_before_digital_add(in $)'],alternative='greater')
print("T value of is: ", round(tvalue, 4))
print("p value is: ", round(pvalue,7))

T value of is:  12.0907
p value is:  0.0


In [16]:
if pvalue<0.05:
    print("reject null hypothesis")
else:
    print("accept null hypothesis")

reject null hypothesis


- Insight : As the p value is less than 0.05 we are rejecting the null hypothesis, and it means that there is a significant increase in the sales after the digital marketing

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

- We need to run a Chi square test for dependency to check.

- Ho = “Region” and “Manager” are independent each other
- Ha = “Region” and “Manager” are dependent each other

In [13]:
# Creating cross tab
data1 = pd.crosstab(data.Region, data.Manager, margins=True)

In [12]:
data1

Manager     Manager - A  Manager - B  Manager - C  All
Region                                                
Region - A            4            3            3   10
Region - B            4            1            2    7
Region - C            1            3            1    5
All                   9            7            6   22

In [24]:
#Finding the P value
data2 = np.array([data1.iloc[0][0:3].values,
                  data1.iloc[1][0:3].values,
                 data1.iloc[2][0:3].values])
stat, p, dof, expected=chi2_contingency(data2)
print("The p value is: ", round(p,4))

The p value is:  0.5494


In [22]:
if p<0.05:
    print("reject null hypothesis")
else:
    print("accept null hypothesis")

accept null hypothesis


- Insight :  As the p value is 0.5494 which is greater than 0.05 and we are accepting the null hypothesis and it means that “Region” and “Manager” are independent each other.